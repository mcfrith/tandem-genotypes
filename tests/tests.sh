#! /bin/sh

try () {
    echo TEST "$@"
    eval "$@"
    echo
}

cd $(dirname $0)

PATH=..:$PATH

{
    try tandem-genotypes --help
    try tandem-genotypes microsat.txt nano.maf
    try tandem-genotypes -s2 -g refFlat.txt simpleRepeat.txt nano.maf
    try tandem-genotypes -s2 -v -g refFlat.txt simpleRepeat.txt nano.maf
    try tandem-genotypes -s2 -g refGene.txt rmsk.txt nano.maf
    try tandem-genotypes -s2 -g refFlat.txt rmsk.out nano.maf
    try tandem-genotypes -s0 -g refFlat.txt simpleRepeat.txt nano.maf
    try tandem-genotypes -s1 -g refFlat.txt simpleRepeat.txt nano.maf
    try tandem-genotypes --postmask=0 near-beg.bed near-beg.maf
    try tandem-genotypes --postmask=0 small-rep.bed small-rep.maf
    try tandem-genotypes --postmask=0 far.bed far.maf
    try tandem-genotypes -u1 -s1 -g refFlat.txt simpleRepeat.txt nano.maf
    try tandem-genotypes -u3 -s2 -g refFlat.txt simpleRepeat.txt nano.maf
    try tandem-genotypes --postmask=0 slop3.bed slop3.maf
    try tandem-genotypes -s2 -p900 -g refFlat.txt simpleRepeat.txt nano.maf
    try tandem-genotypes --postmask=0 ../hg38-disease-tr.txt far.maf
    try "sed 's/chr//' far.maf | tandem-genotypes --postmask=0 far.bed"

    try tandem-genotypes-join --help
    try tandem-genotypes-join min.txt : pro.txt ape.txt
} 2>&1 |
diff -u $(basename $0 .sh).out -
