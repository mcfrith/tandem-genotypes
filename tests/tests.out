TEST tandem-genotypes --help
Usage: tandem-genotypes [options] microsat.txt alignments.maf

Try to indicate genotypes of tandem repeats.

Options:
  -h, --help            show this help message and exit
  -g FILE, --genes=FILE
                        read genes from a genePred or BED file
  -p BP, --promoter=BP  promoter length (default=300)
  -s N, --select=N      select: all repeats (0), non-intergenic repeats (1),
                        non-intergenic non-intronic repeats (2) (default=2)
  -u BP, --min-unit=BP  ignore repeats with unit shorter than BP (default=2)
  --mode=LETTER         L=lenient, S=strict (default=L)
  -v, --verbose         show more details

TEST tandem-genotypes microsat.txt nano.maf
# tandem-genotypes microsat.txt nano.maf
chr22	41994883	41994923	TG	.	.	-7,-7,-4,-3,-1,0,0,0,1,2,3	-16,-14,-13,-13,-13,-12,-12,-11,-9,-8,-7

TEST tandem-genotypes -g refFlat.txt simpleRepeat.txt nano.maf
# tandem-genotypes -g refFlat.txt simpleRepeat.txt nano.maf
chr22	41926181	41926216	CCAGGACCAGCG	TNFRSF13C	coding	-1,0,0,0,0,0,0	0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41958483	41958508	GA	LINC00634	exon	-10,-9,-8,-8,-8,-8,-8,-4	-4,-4,-3,-3,-3,-2,-2,-2,-2,-2,-2,-1,-1,-1,-1,0,1
chr22	41994883	41994923	TG	SEPT3	3'UTR	-7,-7,-4,-3,-1,0,0,0,1,2,3	-16,-14,-13,-13,-13,-12,-12,-11,-9,-8,-7

TEST tandem-genotypes -v -g refFlat.txt simpleRepeat.txt nano.maf
reading refFlat.txt...
reading simpleRepeat.txt...
reading nano.maf...
# tandem-genotypes -v -g refFlat.txt simpleRepeat.txt nano.maf
chr22	41926181	41926216	CCAGGACCAGCG	TNFRSF13C	coding	-1:84868110.682860,0:19064779.75931,0:216722908.247454,0:216722908.458864,0:288286712.341693,0:3976885577.45859,0:4045668814.202393	0:216722908.191156,0:216722908.374159,0:216722908.3872,0:288286712.18532,0:288418386.463180,0:353303576.423872,0:353303576.610171,0:356443753.208271,0:3843483077.135505,0:4045668814.121505,0:4045668814.149539,0:4045668814.441084,0:4045668814.507011,0:4179682758.115941,0:4179682758.29175,0:4179682758.476264,0:4239353418.540835,0:4245879798.88099,0:551111640.357287,0:622291475.174944,0:622291475.574473,0:87746129.42189
chr22	41958483	41958508	GA	LINC00634	exon	-10:3843483077.384706,-9:152889212.45112,-8:152889212.152558,-8:352384898.298207,-8:3709921973.402391,-8:3775529215.186587,-8:622291475.287491,-4:2567311907.26716	-4:3574887596.301650,-4:551111640.334841,-3:353303576.124645,-3:4244727060.265479,-3:551111640.466329,-2:216722908.102623,-2:2567311907.241143,-2:288286712.467921,-2:481119249.119161,-2:622291475.27467,-2:84868110.286543,-1:4239353418.118756,-1:4246400039.335031,-1:4246400039.371226,-1:622291475.156702,0:4045668814.403445,1:86567043.39260
chr22	41994883	41994923	TG	SEPT3	3'UTR	-7:2567311907.307357,-7:551111640.524077,-4:4239353418.326057,-3:288418386.235196,-1:3843483077.2404,0:4239353418.317926,0:4245879798.221634,0:84868110.417637,1:216722908.333390,2:216722908.1709,3:622291475.142691	-16:4179682758.397431,-14:4239353418.504478,-13:4045668814.183140,-13:4045668814.609745,-13:4111103328.245801,-12:19064779.287582,-12:4045668814.734681,-11:356443753.4648,-9:288286712.337975,-8:4246400039.382250,-7:288418386.391961

TEST tandem-genotypes -g refGene.txt rmsk.txt nano.maf
# tandem-genotypes -g refGene.txt rmsk.txt nano.maf
chr22	41914573	41914611	GCGCGA	SHISA8	coding	-1,-1,0,0,0,0,0,0,0,0	-1,0,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41926170	41926217	CACCAG	TNFRSF13C	coding	-3,-1,-1,0,0,0,0	-1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1
chr22	41957928	41957963	CGCC	LINC00634	exon	-4,-3,-2,-2,-1,0,0,0,0,0	-2,-1,-1,-1,-1,-1,-1,-1,0,0,0,0,0,0,0,0,1,2
chr22	41958483	41958508	GA	LINC00634	exon	-10,-9,-8,-8,-8,-8,-8,-4	-4,-4,-3,-3,-3,-2,-2,-2,-2,-2,-2,-1,-1,-1,-1,0,1
chr22	41994883	41994923	TG	SEPT3	3'UTR	-7,-7,-4,-3,-1,0,0,0,1,2,3	-16,-14,-13,-13,-13,-12,-12,-11,-9,-8,-7
chr22	41997631	41997667	CAAA	SEPT3	3'UTR	-2,-1,-1,-1,-1,-1,-1,0,0,0,0	0,0,0,0,0,0,0,1,2,2
chr22	41997742	41997789	CTTCCCC	SEPT3	3'UTR	-2,-1,-1,0,0,0,0,0,0	-1,-1,0,0,0,0,0,0,0,0

TEST tandem-genotypes -g refFlat.txt rmsk.out nano.maf
# tandem-genotypes -g refFlat.txt rmsk.out nano.maf
chr22	41914573	41914611	GCGCGA	SHISA8	coding	-1,-1,0,0,0,0,0,0,0,0	-1,0,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41926170	41926217	CACCAG	TNFRSF13C	coding	-3,-1,-1,0,0,0,0	-1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1
chr22	41957928	41957963	CGCC	LINC00634	exon	-4,-3,-2,-2,-1,0,0,0,0,0	-2,-1,-1,-1,-1,-1,-1,-1,0,0,0,0,0,0,0,0,1,2
chr22	41958483	41958508	GA	LINC00634	exon	-10,-9,-8,-8,-8,-8,-8,-4	-4,-4,-3,-3,-3,-2,-2,-2,-2,-2,-2,-1,-1,-1,-1,0,1
chr22	41994883	41994923	TG	SEPT3	3'UTR	-7,-7,-4,-3,-1,0,0,0,1,2,3	-16,-14,-13,-13,-13,-12,-12,-11,-9,-8,-7
chr22	41997631	41997667	CAAA	SEPT3	3'UTR	-2,-1,-1,-1,-1,-1,-1,0,0,0,0	0,0,0,0,0,0,0,1,2,2
chr22	41997742	41997789	CTTCCCC	SEPT3	3'UTR	-2,-1,-1,0,0,0,0,0,0	-1,-1,0,0,0,0,0,0,0,0

TEST tandem-genotypes -s0 -g refFlat.txt simpleRepeat.txt nano.maf
# tandem-genotypes -s0 -g refFlat.txt simpleRepeat.txt nano.maf
chr22	41915069	41915109	GCTCCG	.	intergenic	-1,0,0,0,0,0,0,1	-2,-1,0,0,0,0,0,0,0,0,0,0
chr22	41916394	41916426	ACACACACATGT	.	intergenic	-1,0,0,0,0,0,0,0,0,0,0	0,0,0,0,0,0,0,0,0,0,0
chr22	41916925	41916952	AAACAAAAAAAAAA	.	intergenic	-1,-1,-1,-1,-1,-1,-1,-1,-1,0,1	-2,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1
chr22	41922641	41922755	AGATGGGGTGGCCAGGAGA	.	intergenic	0,0,0,0,0,0,0,0	-1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41922713	41922797	AGATGGGGTGGG	.	intergenic	0,0,0,0,0,0,0,0	-1,-1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1
chr22	41926181	41926216	CCAGGACCAGCG	TNFRSF13C	coding	-1,0,0,0,0,0,0	0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41929081	41929146	GGTGTGGGTGAGCGTG	.	intergenic	0,0,0,0,0,0,0,0,0,0,0,0	-1,0,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41939778	41939907	AAAG	CENPM	intron	-29,141,148,151	156,192,209,225,225,226,229,234,241,246,267
chr22	41958483	41958508	GA	LINC00634	exon	-10,-9,-8,-8,-8,-8,-8,-4	-4,-4,-3,-3,-3,-2,-2,-2,-2,-2,-2,-1,-1,-1,-1,0,1
chr22	41959510	41959535	AAACA	.	intergenic	-1,0,0,0,0,0,0,1	-1,-1,-1,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41961159	41961203	TTTA	.	intergenic	-2,-2,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,0	-1,-1,-1,-1,-1,-1,-1,0,0,0,0,0,0,0
chr22	41961665	41961701	TTTTTTTTCTGTTTTTTG	.	intergenic	-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1	-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1
chr22	41962342	41962394	TTTCCTTTTTTTTTTTTTCTTTTTTCT	.	intergenic	-2,-2,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1	-2,-2,-2,-2,-2,-2,-2,-2,-2,-1,-1,-1,-1,-1
chr22	41978893	41978931	GGA	SEPT3	intron	-9,-7,-5,-3,-3,-3,-1,0,1	-2,-2,-1,0,0,1,1,1,3
chr22	41982508	41982533	AAAACAA	SEPT3	intron	-1,0,0,0,0,0,0,0,0,0,0,0	-2,0,0,0,0,0,0,0,0
chr22	41991062	41991093	CAAA	SEPT3	intron	-1,-1,-1,0,0,0,0,0,0,0,0,0,0,1,1	-1,-1,-1,0,0,0,0,0,0
chr22	41994883	41994923	TG	SEPT3	3'UTR	-7,-7,-4,-3,-1,0,0,0,1,2,3	-16,-14,-13,-13,-13,-12,-12,-11,-9,-8,-7

TEST tandem-genotypes -s1 -g refFlat.txt simpleRepeat.txt nano.maf
# tandem-genotypes -s1 -g refFlat.txt simpleRepeat.txt nano.maf
chr22	41926181	41926216	CCAGGACCAGCG	TNFRSF13C	coding	-1,0,0,0,0,0,0	0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41939778	41939907	AAAG	CENPM	intron	-29,141,148,151	156,192,209,225,225,226,229,234,241,246,267
chr22	41958483	41958508	GA	LINC00634	exon	-10,-9,-8,-8,-8,-8,-8,-4	-4,-4,-3,-3,-3,-2,-2,-2,-2,-2,-2,-1,-1,-1,-1,0,1
chr22	41978893	41978931	GGA	SEPT3	intron	-9,-7,-5,-3,-3,-3,-1,0,1	-2,-2,-1,0,0,1,1,1,3
chr22	41982508	41982533	AAAACAA	SEPT3	intron	-1,0,0,0,0,0,0,0,0,0,0,0	-2,0,0,0,0,0,0,0,0
chr22	41991062	41991093	CAAA	SEPT3	intron	-1,-1,-1,0,0,0,0,0,0,0,0,0,0,1,1	-1,-1,-1,0,0,0,0,0,0
chr22	41994883	41994923	TG	SEPT3	3'UTR	-7,-7,-4,-3,-1,0,0,0,1,2,3	-16,-14,-13,-13,-13,-12,-12,-11,-9,-8,-7

TEST tandem-genotypes near-beg.bed near-beg.maf
# tandem-genotypes near-beg.bed near-beg.maf
ref	21	146	GGGGCC	.	.	21	.

TEST tandem-genotypes small-rep.bed small-rep.maf
# tandem-genotypes small-rep.bed small-rep.maf

TEST tandem-genotypes far.bed far.maf
# tandem-genotypes far.bed far.maf
chr3	63912686	63912715	CAG	.	.	64	.

TEST tandem-genotypes -u1 -s1 -g refFlat.txt simpleRepeat.txt nano.maf
# tandem-genotypes -u1 -s1 -g refFlat.txt simpleRepeat.txt nano.maf
chr22	41926181	41926216	CCAGGACCAGCG	TNFRSF13C	coding	-1,0,0,0,0,0,0	0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41939778	41939907	AAAG	CENPM	intron	-29,141,148,151	156,192,209,225,225,226,229,234,241,246,267
chr22	41953740	41953768	T	LINC00634	intron	-25,-23,-23,-23,-23,-23,-23,-23,-23,-23,-23,-22,-20	-24,-23,-23,-23,-23,-23,-23,-23,-23,-22,-22,-21
chr22	41957181	41957208	T	LINC00634	intron	-25,-24,-23,-22,-22,-22,-22,-22,-22,-22,-22,-20	-23,-22,-22,-22,-22,-22,-22,-22,-22,-22,-22,-22,-22,-22,-21,-20,-20
chr22	41958483	41958508	GA	LINC00634	exon	-10,-9,-8,-8,-8,-8,-8,-4	-4,-4,-3,-3,-3,-2,-2,-2,-2,-2,-2,-1,-1,-1,-1,0,1
chr22	41978893	41978931	GGA	SEPT3	intron	-9,-7,-5,-3,-3,-3,-1,0,1	-2,-2,-1,0,0,1,1,1,3
chr22	41980129	41980155	T	SEPT3	intron	-21,-21,-21,-21,-21,-21,-20,-20,-20	-24,-23,-22,-22,-21,-21,-19,-19
chr22	41982508	41982533	AAAACAA	SEPT3	intron	-1,0,0,0,0,0,0,0,0,0,0,0	-2,0,0,0,0,0,0,0,0
chr22	41984845	41984870	T	SEPT3	intron	-24,-21,-21,-20,-20,-20,-20,-20,-20,-20,-20,-20,-20,-20,-19,-18,-18	-25,-22,-21,-21,-21,-20,-20,-20,-20,-20,-20,-20,-20,-19,-18
chr22	41991062	41991093	CAAA	SEPT3	intron	-1,-1,-1,0,0,0,0,0,0,0,0,0,0,1,1	-1,-1,-1,0,0,0,0,0,0
chr22	41994883	41994923	TG	SEPT3	3'UTR	-7,-7,-4,-3,-1,0,0,0,1,2,3	-16,-14,-13,-13,-13,-12,-12,-11,-9,-8,-7

TEST tandem-genotypes -u3 -g refFlat.txt simpleRepeat.txt nano.maf
# tandem-genotypes -u3 -g refFlat.txt simpleRepeat.txt nano.maf
chr22	41926181	41926216	CCAGGACCAGCG	TNFRSF13C	coding	-1,0,0,0,0,0,0	0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0

TEST tandem-genotypes slop3.bed slop3.maf
# tandem-genotypes slop3.bed slop3.maf
chr9	27573529	27573546	GGGGCC	.	.	.	16

TEST tandem-genotypes -p900 -g refFlat.txt simpleRepeat.txt nano.maf
# tandem-genotypes -p900 -g refFlat.txt simpleRepeat.txt nano.maf
chr22	41915069	41915109	GCTCCG	SHISA8	promoter	-1,0,0,0,0,0,0,1	-2,-1,0,0,0,0,0,0,0,0,0,0
chr22	41926181	41926216	CCAGGACCAGCG	TNFRSF13C	coding	-1,0,0,0,0,0,0	0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
chr22	41958483	41958508	GA	LINC00634	exon	-10,-9,-8,-8,-8,-8,-8,-4	-4,-4,-3,-3,-3,-2,-2,-2,-2,-2,-2,-1,-1,-1,-1,0,1
chr22	41994883	41994923	TG	SEPT3	3'UTR	-7,-7,-4,-3,-1,0,0,0,1,2,3	-16,-14,-13,-13,-13,-12,-12,-11,-9,-8,-7

TEST tandem-genotypes ../hg38-disease-tr.txt far.maf
# tandem-genotypes ../hg38-disease-tr.txt far.maf
chr3	63912684	63912716	GCA	ATXN7:SCA7	coding	64	.

